### ISSUE TITLE
RI handling fails on Windows

### ISSUE THREAD

**Issue by mih** [datalad/datalad#2545]
It should have 

```
assert_raises(ValueError, clone, 'Zoidberg', path='ssh://mars:Zoidberg')
```

But did:

```
======================================================================
380ERROR: datalad.distribution.tests.test_clone.test_invalid_args
381----------------------------------------------------------------------
382Traceback (most recent call last):
383  File "C:\Miniconda35\envs\test-environment\lib\site-packages\nose\case.py", line 198, in runTest
384    self.test(*self.arg)
385  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\utils.py", line 669, in newfunc
386    return t(*(arg + (filename,)), **kw)
387  File "C:\Miniconda35\envs\test-environment\lib\contextlib.py", line 77, in __exit__
388    self.gen.throw(type, value, traceback)
389  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\utils.py", line 1307, in make_tempfile
390    yield filename
391  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\utils.py", line 669, in newfunc
392    return t(*(arg + (filename,)), **kw)
393  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\utils.py", line 669, in newfunc
394    return t(*(arg + (filename,)), **kw)
395  File "C:\Miniconda35\envs\test-environment\lib\contextlib.py", line 77, in __exit__
396    self.gen.throw(type, value, traceback)
397  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\utils.py", line 1307, in make_tempfile
398    yield filename
399  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\utils.py", line 669, in newfunc
400    return t(*(arg + (filename,)), **kw)
401  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\utils.py", line 669, in newfunc
402    return t(*(arg + (filename,)), **kw)
403  File "C:\Miniconda35\envs\test-environment\lib\contextlib.py", line 77, in __exit__
404    self.gen.throw(type, value, traceback)
405  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\utils.py", line 1307, in make_tempfile
406    yield filename
407  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\tests\utils.py", line 669, in newfunc
408    return t(*(arg + (filename,)), **kw)
409  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\distribution\tests\test_clone.py", line 71, in test_invalid_args
410    assert_raises(ValueError, clone, 'Zoidberg', path='ssh://mars:Zoidberg')
411  File "C:\Miniconda35\envs\test-environment\lib\unittest\case.py", line 727, in assertRaises
412    return context.handle('assertRaises', args, kwargs)
413  File "C:\Miniconda35\envs\test-environment\lib\unittest\case.py", line 176, in handle
414    callable_obj(*args, **kwargs)
415  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\interface\utils.py", line 470, in eval_func
416    return return_func(generator_func)(*args, **kwargs)
417  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\interface\utils.py", line 458, in return_func
418    results = list(results)
419  File "C:\Miniconda35\envs\test-environment\lib\site-packages\datalad\interface\utils.py", line 446, in generator_func
420    msg="Command did not complete successfully")
421datalad.support.exceptions.IncompleteResultsError: Command did not complete successfully [{'status': 'error', 'type': 'dataset', 'source_url': 'Zoidberg', 'message': ('Failed to clone data from any candidate source URL: %s', ['Zoidberg']), 'action': 'install', 'path': 'C:\\projects\\datalad-5sgef\\__testhome__\\ssh:\\mars:Zoidberg'}]
```

**Comment by bpoldrack** [comment]
FTR: At a first glance, I suspect the colon in `C:\` to cause a parsing issue here ...

**Comment by mih** [comment]
Ah, no. The input is `ssh://mars:Zoidberg` not `c:\` in it.

**Comment by bpoldrack** [comment]
Sure. But I recall sth like first trying to interpret it as a path, which then would lead to assuming a relative path (since it's obviously not absolute), therefore trying to convert to sth like `C:\somewhere\ssh://mars:Zoidberg`. Something along these lines ... Of course my remembrance my be entirely wrong or simply outdated.

**Comment by bpoldrack** [comment]
I'm afraid I'm pretty close with the guess above. `clone` does sth very similar and is NOT using `RI` capabilities first.

**Comment by bpoldrack** [comment]
Actually, I start wondering why the hell it raises on linux ;-)
Anyways - I'm on it.

**Comment by bpoldrack** [comment]
Indeed: Even on linux, `clone` concludes the target to be sth like `/tmp/datalad-test/ssh:/mars:Zoidberg`.
What's causing the expected Exception is in fact the `Dataset` constructor, not some `RI` stuff. It should fail before and use `RI` to make sure, `path` is in fact a `PathRI`. I'll fix that.

**Comment by mih** [comment]
Any progress? After enabling SSH tests on windows the issue reveals itself as rather wide-spread:

https://ci.appveyor.com/project/mih/datalad-5sgef/build/1.0.91

**Comment by bpoldrack** [comment]
Kind of.

```
In[5]: RI("/home/ben/ssh://mars:zoidberg")
Out[5]: SSHRI(hostname='/home/ben/ssh', path='/mars:zoidberg')
```
vs.
```
In[7]: RI("C:\home\ben\ssh://mars:zoidberg")
Out[7]: PathRI(path='C:\\home\x08en\\ssh://mars:zoidberg')
```

The issue is twofold. For one `clone` does convert the argument to an absolute path without first checking whether it is an `PathRI`. On linux it still works, since `Dataset()` then calls `RI` which still recognizes that "path" as an `SSHRI` (although it's actually not a valid one anymore). On Windows the `C:\` at the beginning is sufficient to let `RI` detect a `PathRI` instead.

**Comment by bpoldrack** [comment]
Regarding other similar failures, there's even more to it.
In the build you linked there's `clone1.create_sibling('ssh://localhost%s/%s' % (getpwd(), clone2_), name=clone2_)`, which then fails at the end trying to interpret `\Users\appveyor\AppData\Local\Temp\1\datalad_temp_65tyxeij` as part.
So, we build `ssh://localhostC:\Users\appveyor\AppData\Local\Temp\1\datalad_temp_65tyxeij` which is wrong of course. I'm just not sure how exactly the correct SSH URL needs to look like. However, that's a different issue, resulting from us using `getcwd` as something we can put into such a URL - that's not `RI`s fault.

**Comment by bpoldrack** [comment]
So, my little fix solved it for the `clone` tests. Tried to solve as generally as possible by letting `dataset.py:resolve_path()` check, whether it actually got a `PathRI` to deal with instead of returning such nonsense as `C:\something\ssh://whatever`.

Looking into valid SSH URL with paths on windows now ...

**Comment by bpoldrack** [comment]
The RI issues in your above linked build, complaining about a port `\User\some\where` seems to be solved in PR #2556 as well. For comparison: https://ci.appveyor.com/project/mih/datalad/build/1.0.80

The failures are now all about a not existing `SSHManager`, which is a different issue.
